### PR TITLE
[improvement](auth)select priv for Nereids

### PR DESCRIPTION
**Pull request by zddr**
## Proposed changes

Issue Number: close #xxx

New optimizer select permission supports catalog

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/UserAuthentication.java [hunk 1]
```diff
@@ -55,14 +56,16 @@ private Plan checkPermission(LogicalRelation relation, ConnectContext connectCon
         }
 
         if (relationsToCheck.contains(relation.getClass())) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/UserAuthentication.java:58]: do we need update relationsToCheck list?

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/UserAuthentication.java [hunk 2]
```diff
@@ -55,14 +56,16 @@ private Plan checkPermission(LogicalRelation relation, ConnectContext connectCon
...
-            String dbName =
-                    !relation.getQualifier().isEmpty() ? relation.getQualifier().get(0) : null;
-            String tableName = relation.getTable().getName();
-            if (!connectContext.getEnv().getAccessManager().checkTblPriv(connectContext, dbName,
+            TableIf table = relation.getTable();
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/UserAuthentication.java:59]: not all relation has table, so u need to check null here

**Review by morningman — COMMENTED** [review]
please add regression test

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**